### ISSUE TITLE
JWT::decode($tokenString, $this->firebaseJwtKey); breaks

### ISSUE THREAD

**Issue by SpacePossum** [firebase/php-jwt#454]
Hi all,
thanks for maintaining and publishing this package, much appreciated :)

There is a an issue with the `JWT::decode`. 
In `JWT::decode` it does a call like:

```php
$key = self::getKey($keyOrKeyArray, property_exists($header, 'kid') ? $header->kid : null);
```
(https://github.com/firebase/php-jwt/blob/v6.3.0/src/JWT.php#L128)

Its signature:

```php
    private static function getKey(
        $keyOrKeyArray,
        ?string $kid
    ): Key {
```

(https://github.com/firebase/php-jwt/blob/v6.3.0/src/JWT.php#L420)

In this method we have:

```php
    if ($keyOrKeyArray instanceof CachedKeySet) {
        // Skip "isset" check, as this will automatically refresh if not set
        return $keyOrKeyArray[$kid];
    }

    if (empty($kid)) {
```

note the order of things:
- first we use the `$kid` even if it is `null` as long as `$keyOrKeyArray instanceof CachedKeySet`
- only if this is not true we enter `if (empty($kid))`

As such if we have `$keyOrKeyArray instanceof CachedKeySet` and `$kid` is `null` we try to return:

```php
    return $keyOrKeyArray[$kid];
```

Looking into `CachedKeySet`, this `keyOrKeyArray[$kid];` calls;

```php
    public function offsetGet($keyId): Key
```

(https://github.com/firebase/php-jwt/blob/main/src/CachedKeySet.php#L99)

from which than we call this private function 

```php
    private function keyIdExists(string $keyId): bool`
```

(https://github.com/firebase/php-jwt/blob/main/src/CachedKeySet.php#L133)

Since `$keyId` is `null` calling `keyIdExists` with type hint `(string $keyId)` breaks.

To resolve the bug I think changing the type hint to `private function keyIdExists(?string $keyId): bool` would work, however I am not (very) familiar with this project.

**Comment by bshaffer** [comment]
I agree with your assessment. Thank you for catching this, and for your detailed report!

**Comment by bshaffer** [comment]
fixed in bad1b040d0c736bbf86814c6b5ae614f517cf7bd

**Comment by SpacePossum** [comment]
Thanks @bshaffer :)  much appreciated :+1:   looking forward to the next release :rocket:
